tests/specs/exports-types.ts

```diff
@@ -1,206 +1,209 @@
 import { testSuite, expect } from 'manten';
-import { createPackage, createPkgJson } from '../utils.js';
-import { getPackageEntryPoints } from '#pkg-entry-points';
+import { createPackage, createPkgJson, testScenarios } from '../utils.js';
 
 export default testSuite(({ describe }) => {
-	describe('exports =', ({ test, describe }) => {
-		test('string', async () => {
-			const {
-				fixture,
-				packagePath,
-				assertSubpath,
-			} = await createPackage({
-				pkg: {
-					'package.json': createPkgJson({
-						exports: './file.mjs',
-					}),
-					'file.mjs': 'export default 123',
-				},
-			});
-
-			await assertSubpath('pkg', []);
-
-			const packageExports = await getPackageEntryPoints(packagePath);
-			expect(packageExports).toStrictEqual({
-				'.': [
-					[['default'], './file.mjs'],
-				],
-			});
-
-			await fixture.rm();
-		});
-
-		test('null', async () => {
-			const {
-				fixture,
-				packagePath,
-				assertSubpath,
-			} = await createPackage({
-				pkg: {
-					'package.json': createPkgJson({
-						exports: null,
-					}),
-					'file.mjs': 'export default 123',
-				},
-			});
-
-			await expect(() => assertSubpath('pkg', [])).rejects.toThrowError('ERR_MODULE_NOT_FOUND');
-
-			const packageExports = await getPackageEntryPoints(packagePath);
-			expect(packageExports).toStrictEqual({});
-
-			await fixture.rm();
-		});
-
-		describe('conditions object', ({ test }) => {
-			test('conditions object', async () => {
-				const {
-					fixture,
-					packagePath,
-					assertSubpath,
-				} = await createPackage({
-					pkg: {
-						'package.json': createPkgJson({
-							exports: {
-								conditionA: './file.mjs',
-								'condition-b': './file.mjs',
-							},
-						}),
-						'file.mjs': 'export default 123',
-					},
+	for (const { scenario, getPackageEntryPoints } of testScenarios) {
+		describe(scenario, ({ describe }) => {
```

Comment:
This is running the existing test suites in two scenarios: async + sync. The rest of the diff is just the change in indentation, so best reviewed with "hide whitespace" enabled!